### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by nulltoken**

### CODE REVIEW

File: LibGit2Sharp/Repository.cs [hunk 1]
```diff
@@ -565,6 +565,8 @@ public static string Discover(string startingPath)
                     repoPath = Proxy.git_repository_path(repo);
                 }
 
+                EncodingMarshaler.Cleanup(cloneOpts.CheckoutBranch);
```
Comment (jamill) [inline LibGit2Sharp/Repository.cs]: I think we want this cleanup to run even in the clone throws an exception? Should we put this in a try / finally block?

Comment (nulltoken) [inline LibGit2Sharp/Repository.cs]: You're right!

Comment (nulltoken) [inline LibGit2Sharp/Repository.cs]: Fixed. Better?

**Comment by Therzok** [comment]
:+1:

**Comment by jamill** [comment]
:+1:

**Comment by nulltoken** [comment]
@Therzok Would there be a way to valgrind the linux build (if that makes sense)?

**Comment by Therzok** [comment]
By that, you mean using valgrind on `mono`? Should be possible. You could do run the xunit console runner with the .NET 4 mono by putting it under valgrind. Would've done it on OS X, but Valgrind's dead in Yosemite.

**Pull request merged**